File: clients/ios/Lextures/Core/LMS/GamificationLogic.swift
import Foundation

/// Gamification display and privacy helpers (M9.3).
enum GamificationLogic {
    static func gamificationEnabled(_ features: MobilePlatformFeatures) -> Bool {
        features.ffGamification
    }

    static func cacheKey() -> String { "gamification:profile" }

    static func leaderboardCacheKey(courseCode: String) -> String {
        "gamification:leaderboard:\(courseCode)"
    }

    static func badgeLabel(_ badgeType: String) -> String {
        switch badgeType {
        case "streak_7":
            return L.text("mobile.gamification.badge.streak7")
        case "streak_30":
            return L.text("mobile.gamification.badge.streak30")
        case "xp_100":
            return L.text("mobile.gamification.badge.xp100")
        case "xp_1000":
            return L.text("mobile.gamification.badge.xp1000")
        case "first_course_complete":
            return L.text("mobile.gamification.badge.firstCourse")
        default:
            let spaced = badgeType.replacingOccurrences(of: "_", with: " ")
            return spaced.prefix(1).uppercased() + spaced.dropFirst()
        }
    }

    static func levelProgressLabel(profile: GamificationProfile) -> String {
        L.format(
            "mobile.gamification.levelProgress",
            profile.level,
            profile.xpToNextLevel,
            Int(profile.levelProgressPct.rounded())
        )
    }

    static func shouldShowLeaderboard(profile: GamificationProfile) -> Bool {
        profile.leaderboardVisible
    }

    static func leaderboardOptOutMessage() -> String {
        L.text("mobile.gamification.leaderboardOptOut")
    }

    static func canUseStreakFreeze(profile: GamificationProfile) -> Bool {
        profile.streakFreezes > 0 && profile.currentStreak > 0
    }

    static func streakRiskMessage(profile: GamificationProfile) -> String? {
        guard profile.streakAtRisk else { return nil }
        if let hours = profile.streakHoursLeft, hours > 0 {
            return L.format("mobile.gamification.streakAtRiskHours", Int(hours.rounded(.up)))
        }
        return L.text("mobile.gamification.streakAtRisk")
    }
}

File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/GamificationLogic.kt
package com.lextures.android.core.lms

import com.lextures.android.core.navigation.MobilePlatformFeatures

object GamificationLogic {
    fun gamificationEnabled(features: MobilePlatformFeatures): Boolean = features.ffGamification

    fun cacheKey(): String = "gamification:profile"

    fun leaderboardCacheKey(courseCode: String): String = "gamification:leaderboard:$courseCode"

    fun badgeLabel(badgeType: String): String = when (badgeType) {
        "streak_7" -> "7-day streak"
        "streak_30" -> "30-day streak"
        "xp_100" -> "100 XP"
        "xp_1000" -> "1000 XP"
        "first_course_complete" -> "First course complete"
        else -> badgeType.replace('_', ' ').replaceFirstChar { it.uppercase() }
    }

    fun shouldShowLeaderboard(profile: GamificationProfile): Boolean = profile.leaderboardVisible

    fun canUseStreakFreeze(profile: GamificationProfile): Boolean =
        profile.streakFreezes > 0 && profile.currentStreak > 0
}